tests/system/src/test/java/test/util/ScreenCaptureTestWatcher.java

```diff
@@ -40,9 +45,13 @@
  * Make sure there is plenty of free disk space.
  */
 public class ScreenCaptureTestWatcher implements TestWatcher {
+
+    private static final String PROPERTY = "test.screenshot";
+
     @Override
     public void testFailed(ExtensionContext extensionContext, Throwable err) {
```

Comment:
If i induce failure in `test.robot.javafx.scene.layout.RegionBackgroundFillUITest.basicFill_Radius4` and run it with `@ExtendWith(ScreenCaptureTestWatcher.class)` annotation on RegionBackgroundFillUITest class. I see that screen capture happens after the test window in closed.

May be in such cases, its better to call `ScreenshotCapture.writeScreenshot()` directly before we throw AssertionError on failure?